TrueCraft.Client/Rendering/Blocks/SlabRenderer.cs

```diff
@@ -0,0 +1,235 @@
+            WoodTopTexture + Vector2.UnitX + Vector2.UnitY,
+            WoodTopTexture + Vector2.UnitY,
+            WoodTopTexture,
+            WoodTopTexture + Vector2.UnitX,
+            // Negative Y
+            WoodBottomTexture + Vector2.UnitX + Vector2.UnitY,
+            WoodBottomTexture + Vector2.UnitY,
+            WoodBottomTexture,
+            WoodBottomTexture + Vector2.UnitX,
+        };
+
+        private static Vector2[] CobbleTextureMap =
+        {
+            // Positive Z
+            CobbleSideTexture + Vector2.UnitX + Vector2.UnitY,
+            CobbleSideTexture + Vector2.UnitY,
+            CobbleSideTexture,
+            CobbleSideTexture + Vector2.UnitX,
+            // Negative Z
+            CobbleSideTexture + Vector2.UnitX + Vector2.UnitY,
+            CobbleSideTexture + Vector2.UnitY,
+            CobbleSideTexture,
+            CobbleSideTexture + Vector2.UnitX,
+            // Positive X
+            CobbleSideTexture + Vector2.UnitX + Vector2.UnitY,
+            CobbleSideTexture + Vector2.UnitY,
+            CobbleSideTexture,
+            CobbleSideTexture + Vector2.UnitX,
+            // Negative X
+            CobbleSideTexture + Vector2.UnitX + Vector2.UnitY,
+            CobbleSideTexture + Vector2.UnitY,
+            CobbleSideTexture,
+            CobbleSideTexture + Vector2.UnitX,
+            // Negative Y
+            CobbleTopTexture + Vector2.UnitX + Vector2.UnitY,
+            CobbleTopTexture + Vector2.UnitY,
+            CobbleTopTexture,
+            CobbleTopTexture + Vector2.UnitX,
+            // Negative Y
+            CobbleBottomTexture + Vector2.UnitX + Vector2.UnitY,
+            CobbleBottomTexture + Vector2.UnitY,
+            CobbleBottomTexture,
+            CobbleBottomTexture + Vector2.UnitX,
+        };
+
+        static SlabRenderer()
+        {
+            BlockRenderer.RegisterRenderer(SlabBlock.BlockID, new SlabRenderer());
+            BlockRenderer.RegisterRenderer(DoubleSlabBlock.BlockID, new SlabRenderer());
+
+            for (int i = 0; i < StoneTextureMap.Length; i++)
+            {
+                StoneTextureMap[i] *= new Vector2(16f / 256f);
+                SandstoneTextureMap[i] *= new Vector2(16f / 256f);
+                WoodTextureMap[i] *= new Vector2(16f / 256f);
+                CobbleTextureMap[i] *= new Vector2(16f / 256f);
+            }
+        }
+
+        protected virtual Vector2[] GetTextureMap(SlabBlock.SlabMaterial material)
+        {
+            switch (material)
+            {
+                case SlabBlock.SlabMaterial.Stone:
+                    return StoneTextureMap;
+                case SlabBlock.SlabMaterial.Standstone:
+                    return SandstoneTextureMap;
+                case SlabBlock.SlabMaterial.Wooden:
+                    return WoodTextureMap;
+                case SlabBlock.SlabMaterial.Cobblestone:
+                    return CobbleTextureMap;
+                default:
+                    return null;
```

Comment:
This will probably lead to a client crash. We should return stone slab textures as the default.